bench/jq/optimized/regexec.ll

```diff
@@ -11243,52 +11243,58 @@ define internal range(i32 -30, 1) i32 @onig_builtin_monitor(ptr nocapture nounde
 
 38:                                               ; preds = %22
   %39 = call i32 (ptr, i64, ptr, ...) @snprintf(ptr noundef nonnull dereferenceable(1) %3, i64 noundef 20, ptr noundef nonnull @.str.2, i32 noundef %23) #29
-  br label %48
+  br label %49
 
 40:                                               ; preds = %22
   %41 = ptrtoint ptr %36 to i64
   %42 = ptrtoint ptr %35 to i64
   %43 = sub i64 %41, %42
   %spec.store.select = tail call i64 @llvm.umin.i64(i64 %43, i64 19)
   %.not = icmp eq ptr %36, %35
-  br i1 %.not, label %._crit_edge, label %.lr.ph
-
-.lr.ph:                                           ; preds = %40, %.lr.ph
-  %indvars.iv = phi i64 [ %indvars.iv.next, %.lr.ph ], [ 0, %40 ]
-  %44 = getelementptr inbounds i8, ptr %35, i64 %indvars.iv
-  %45 = load i8, ptr %44, align 1
-  %46 = getelementptr inbounds [20 x i8], ptr %3, i64 0, i64 %indvars.iv
-  store i8 %45, ptr %46, align 1
+  br i1 %.not, label %._crit_edge, label %.lr.ph.preheader
+
+.lr.ph.preheader:                                 ; preds = %40
+  %44 = trunc nuw nsw i64 %spec.store.select to i32
+  %umax = tail call i32 @llvm.umax.i32(i32 %44, i32 1)
+  %wide.trip.count = zext nneg i32 %umax to i64
```

Comment:
Regression.

Reply:
Duplicate of https://github.com/dtcxzyw/llvm-opt-benchmark/pull/600#discussion_r1597406008